mender-conversion-tool.sh

```diff
@@ -0,0 +1,688 @@
+  echo $sudoPW | sudo -S cp MLO $BASE_DIR/mnt/boot/
+  echo $sudoPW | sudo -S cp u-boot.img $BASE_DIR/mnt/boot/
+  cd $BASE_DIR
+  set_uenv
+}
+
+set_rootfs() {
+  echo "Setting root filesystem..."
+  [ ! -d "$BASE_DIR/mnt/rootfs" ] && { echo "Error: invalid rootfs source."; exit 1; }
+  [ ! -d "$BASE_DIR/mnt/primary" ] && { echo "Error: invalid rootfs target."; exit 1; }
+
+  echo $sudoPW | sudo -S cp -ar $BASE_DIR/mnt/rootfs/* $BASE_DIR/mnt/primary
+  # Add mountpoints.
+  echo $sudoPW | sudo -S install -d -m 755 $BASE_DIR/mnt/primary/uboot
+  echo $sudoPW | sudo -S install -d -m 755 $BASE_DIR/mnt/primary/data
+}
+
+set_fstab() {
+  echo "Setting fstab..."
+  sysconfdir="$BASE_DIR/mnt/primary/etc"
+
+  [ ! -d "${sysconfdir}" ] && { echo "Error: cannot find rootfs config dir."; exit 1; }
+  # Erase/create the fstab file.
+  echo $sudoPW | sudo -S install -b -m 644 /dev/null ${sysconfdir}/fstab
+
+  # Add Mender specific entries to fstab.
+  echo $sudoPW | sudo -S bash -c "cat <<- EOF > ${sysconfdir}/fstab
+	# stock fstab - you probably want to override this with a machine specific one
+
+	/dev/root            /                    auto       defaults              1  1
+	proc                 /proc                proc       defaults              0  0
+	devpts               /dev/pts             devpts     mode=0620,gid=5       0  0
+	tmpfs                /run                 tmpfs      mode=0755,nodev,nosuid,strictatime 0  0
+	tmpfs                /var/volatile        tmpfs      defaults              0  0
+
+	# uncomment this if your device has a SD/MMC/Transflash slot
+	#/dev/mmcblk0p1       /media/card          auto       defaults,sync,noauto  0  0
+
+	# Where the U-Boot environment resides; for devices with SD card support ONLY!
+	/dev/mmcblk0p1   /uboot               auto       defaults,sync    0  0
+	/dev/mmcblk0p4   /data                auto       defaults         0  0
+	EOF"
+}
+
+clean_and_detach() {
+  cd $BASE_DIR
+
+  # In case of troubles with umount, check which process makes mountpoint busy.
+  fuser -vm $BASE_DIR/mnt/rootfs 2>&1 >/dev/null
+  fuser -vm $BASE_DIR/mnt/primary 2>&1 >/dev/null
+
+  echo $sudoPW | sudo -S umount $BASE_DIR/mnt/rootfs
+  [[ $? -ne 0 ]] && { echo $sudoPW | sudo -S umount -l $BASE_DIR/mnt/rootfs; }
+  
+  for i in $(seq 1 $number_of_parts);
+  do
+    echo $sudoPW | sudo -S umount -l $BASE_DIR/mnt/${part_names[${i} - 1]}
+    [[ $? -ne 0 ]] && { echo $sudoPW | sudo -S umount -l $BASE_DIR/mnt/${part_names[${i} - 1]}; }
+  done
+
+  echo $sudoPW | sudo -S losetup $loopsdimg 2>&1 >/dev/null
+  [[ $? -eq 0 ]] && { echo $sudoPW | sudo -S losetup -d $loopsdimg; }
+
+  echo $sudoPW | sudo -S kpartx -d /dev/"$1" &
+  echo $sudoPW | sudo -S losetup -d /dev/"$1" &
+  wait
+  rm -rf $BASE_DIR/mnt
+  echo $sudoPW | sudo -S losetup -a
+}
+
+do_prepare_image() {
+  loopdevice=($(echo $sudoPW | sudo -S losetup -f))
+  echo $sudoPW | sudo -S losetup $loopdevice $image -o $((8192*512))
```

Comment:
What is `8192*512`?

Reply:
New method will be added with the amend commit to analyse a source embedded image. No more hard-coded values.